Fixed #33256 -- Fixed schema test failures when using --keepdb.

likecodingloveproblems opened the pull request:
Some tests leave models and for the second run of test,
model already exist error was created.
So models are deleted at the test it self

charettes commented:
FWIW I think that's what `self.local_models` and `self.isolated_local_models` are for

https://github.com/django/django/blob/cfc7c3f17ce724d682753501652310159fab2eca/tests/schema/tests.py#L128-L133

likecodingloveproblems commented:
Thank you @charettes for your advice, it works nice with `self.isolated_local_models` except `test_unique_name_quoting` test, because of altering table_name `tearDown` try to drop `schema_table`.

I search too much to make a copy of `TagUniqueRename` class, but I can't solve it, do you have any idea?

likecodingloveproblems commented:
@felixxm Thanks for your precise review.
Why you think setting local_model and isolated_local_model most be must be outside of the context processor?

likecodingloveproblems commented on tests/schema/tests.py:
```diff
@@ -2314,6 +2314,8 @@ def test_unique_name_quoting(self):
                 editor.alter_db_table(TagUniqueRename, old_table_name, "unique-table")
                 TagUniqueRename._meta.db_table = "unique-table"
+                TagUniqueRenameCopy = type('TagUniqueRenameCopy', (), {'_meta':copy(TagUniqueRename._meta)})
+                self.isolated_local_models = [TagUniqueRenameCopy]
```
Please leave feedback on my solution, why it was not good enough, if it's possible?

felixxm replied:
> Please leave feedback on my solution, why it was not good enough, if it's possible?

It's unnecessarily complicated.

likecodingloveproblems commented:
@felixxm Why does tests run correctly when I use isolated_local_models instead of local_models?

felixxm commented:
> Why you think setting local_model and isolated_local_model most be must be outside of the context processor?

Adding models to `local_models`/`isolated_local_models` doesn't need to be in the context processor, so there is no reason to keep them there.

likecodingloveproblems commented:
What's the next step?
Must I close the pull request?

felixxm commented:
> What's the next step? Must I close the pull request?

Please don't do anything. I squashed commits and fixed one more case on Oracle.

felixxm commented:
buildbot, test on oracle.

likecodingloveproblems commented:
> > What's the next step? Must I close the pull request?
> 
> Please don't do anything. I squashed commits and fixed one more case on Oracle.

Thanks for your nice support

felixxm commented:
Welcome aboard :boat:
